src/components/ArticlesList.jsx

```diff
@@ -1,29 +1,67 @@
-import React, { useEffect, useState } from "react";
 import axios from "axios";
-import ArticleCard from "./ArticleCard";
-import {useParams} from "react-router-dom"
+import React from "react";
+import { useState, useEffect } from "react";
+import ArticleCard  from "./ArticleCard";
 
-export default function ArticlesList () {
-    const [articlesList, setArticlesList] = useState([])
-    
-    
-    useEffect(() => {
-        axios
-          .get("https://be-nc-news-tnfa.onrender.com/api/articles")
-          .then(({ data }) => {
-            console.log(data.articles)
-            setArticlesList(data.articles);
-          });
-      }, []);
+export default function topicsForm() {
+  const [topicsForm, setTopicsForm] = useState({
+    category_name: "",
+  });
 
-    return (
-        <div className="articles">
-            <h1>Articles:</h1>
-          {articlesList.map((article) => {
-            return <ArticleCard key={article.article_id} article={article} />;
-          })}
-                <br></br>
-        </div>
-      );
+  const [articlesList, setArticlesList] = useState([])
+
+  function handleChange(e) {
+    const { name, value } = e.target;
+    setTopicsForm({
+      ...topicsForm,
+      [name]: value,
+    });
+  const newURL = `http://localhost:5174/articles/topics/${value}`;
+  window.history.pushState(null, "", newURL);
+}
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    getTopics(topicsForm);
+  };
+  console.log(topicsForm)
+
+useEffect(() => {
+  axios
+    .get("https://be-nc-news-tnfa.onrender.com/api/articles")
+    .then(({ data }) => {
+      console.log(data.articles)
+      setArticlesList(data.articles);
+    });
+}, []);
+
+return (
+<>
+<form className="topicsForm" onSubmit={handleSubmit}>
+  <p>filter topics:</p>
+<select
+    value={topicsForm.slug}
+    id="category_name"
+    name="category_name"
+    onChange={handleChange}
+  >
+    <option value="coding">Coding</option>
+    <option value="football">Football</option>
+    <option value="cooking">Cooking</option>
+    <option value="allArticles">allArticles</option>
+  </select>
+</form>
+      {articlesList.map((article) => {
+        if (article.topic === topicsForm.category_name) {
```

Comment:
This frontend logic is filtering by topic. This is something that ideally would be done using the backend query however I don't believe your backend API has that set up. So in place of that this is good logic to implement the feature